Compatibility with Openfire 5.0.0

When loading version 2.1.10 in Openfire 5.0.0, the admin console of the plugin's page is empty (as shown below) and the logs contain a stack trace like this:

```
2025.06.24 10:23:48.419 ERROR [Jetty-QTP-AdminConsole-92]: org.jivesoftware.openfire.container.PluginServlet - java.lang.NoSuchMethodError: 'org.jivesoftware.openfire.container.Plugin org.jivesoftware.openfire.container.PluginManager.getPlugin(java.lang.String)'
javax.servlet.ServletException: java.lang.NoSuchMethodError: 'org.jivesoftware.openfire.container.Plugin org.jivesoftware.openfire.container.PluginManager.getPlugin(java.lang.String)'
        at org.apache.jasper.runtime.PageContextImpl.handlePageException(PageContextImpl.java:633) ~[apache-jsp-9.0.105.jar:9.0.105]
        at org.jivesoftware.openfire.plugin.clientControl.client_002dfeatures_jsp._jspService(client_002dfeatures_jsp.java:978) ~[?:?]
        at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:64) ~[apache-jsp-9.0.105.jar:9.0.105]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:590) ~[jetty-servlet-api-4.0.6.jar:?]
        at org.jivesoftware.openfire.container.PluginServlet.handleJSP(PluginServlet.java:447) ~[xmppserver-5.1.0-SNAPSHOT.jar:5.1.0-SNAPSHOT]
        at org.jivesoftware.openfire.container.PluginServlet.service(PluginServlet.java:107) ~[xmppserver-5.1.0-SNAPSHOT.jar:5.1.0-SNAPSHOT]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:590) ~[jetty-servlet-api-4.0.6.jar:?]
        at org.eclipse.jetty.ee8.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1151) ~[jetty-ee8-servlet-12.0.22.jar:12.0.22]
        at org.eclipse.jetty.ee8.servlet.ServletHolder.handle(ServletHolder.java:640) ~[jetty-ee8-servlet-12.0.22.jar:12.0.22]
... snipped for brevity ...
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1166) ~[?:?]
        at java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: java.lang.NoSuchMethodError: 'org.jivesoftware.openfire.container.Plugin org.jivesoftware.openfire.container.PluginManager.getPlugin(java.lang.String)'
        at org.jivesoftware.openfire.plugin.clientControl.client_002dfeatures_jsp._jspService(client_002dfeatures_jsp.java:360) ~[?:?]
        ... 71 more

```

![Image](https://github.com/user-attachments/assets/2eaf782f-68ca-4d71-82ff-e8ce98cda0e8)

--- a/changelog.html
+++ b/changelog.html
@@ -46,6 +46,7 @@ Client Control Plugin Changelog
 
 <p><b>2.1.10</b> (To be determined)</p>
 <ul>
+    <li>[<a href='https://github.com/igniterealtime/openfire-clientControl-plugin/issues/51'>Issue #51</a>] - Compatible with Openfire 5.0.0</li>
 </ul>
 
 <p><b>2.1.9</b> September 12, 2024</p>

--- a/plugin.xml
+++ b/plugin.xml
@@ -9,7 +9,7 @@
     <description>Controls clients allowed to connect and available features</description>
     <author>Jive Software</author>
     <version>${project.version}</version>
-    <date>2024-09-12</date>
+    <date>2025-06-24</date>
     <minServerVersion>4.8.0</minServerVersion>
 
     <!-- UI extension -->

--- a/src/web/client-features.jsp
+++ b/src/web/client-features.jsp
@@ -226,7 +226,7 @@
 	
     // Enable File Transfer in the system.
     ClientControlPlugin plugin = (ClientControlPlugin) XMPPServer.getInstance()
-            .getPluginManager().getPlugin("clientcontrol");
+            .getPluginManager().getPluginByName("Client Control").orElseThrow();
     FileTransferFilterManager manager = plugin.getFileTransferFilterManager();
     manager.enableFileTransfer(transferEnabled);
 %>

